packages/core/src/babel-plugin-card-schema-transform.ts

```diff
@@ -740,7 +740,7 @@ function addLoadedFieldsMethod(path: NodePath<t.Class>, state: State, babel: typ
       t.identifier('loadedFields'),
       [t.identifier('schemaInstance')],
       t.blockStatement([
-        babel.template(`return [...schemaInstance.%%loadedFields%%];`)({
+        babel.template(`return allFields.length === 0 ? [] : [...schemaInstance.%%loadedFields%%];`)({
```

Comment:
let's fix this in the constructor so that `loadedFields` is always correct. I suspect this is why you added a lot of downstream logic that didn't trust the value of `schemaInstance.loadedFields`

Reply:
i think that is the right thing to do.

Reply:
it should be ok to adjust the tests that thing the constructor should be empty to no longer being so. The primary job of those tests is to make it easier to see what the code that the compiler emits. since we changed the code that the compiler emits, it makes sense those tests that were expecting primitive cards to have no constructor will have to change now.